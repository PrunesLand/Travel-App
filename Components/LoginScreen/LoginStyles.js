import {StyleSheet} from 'react-native'

export const Login = StyleSheet.create({
    loginContainer:{
        // width:'100%',
        // height: '60vh',
        
        
       height:300,
       width:300,
        
    },
    container:{
        flex:1
    },
    logoContainer:{
        flex:1,
        top:25,
        left:15,
        display:'flex',
        justifyContent:'center',
        alignItems:'center',  
        bottom:8
    },
    square:{
        backgroundColor: 'rgba(45, 106, 79, 0.9)',
        
        alignItems:'center',
        borderRadius:15,
        width:370,
        height:350,
        top:100,
        left:10
    }
})