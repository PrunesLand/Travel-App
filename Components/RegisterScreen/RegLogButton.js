import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import { regButton } from './RegisterStyles'

const RegLogButton = ({title, onPress}) => {
    return (
        <TouchableOpacity onPress={onPress}>
            <View style={regButton.Button}>
                <Text style={regButton.register}>
                    {title}
                </Text>
            </View>
        </TouchableOpacity>
    )
}

export default RegLogButton
