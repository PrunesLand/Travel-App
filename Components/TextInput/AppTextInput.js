import React from 'react'
import { View, Text, TextInput } from 'react-native'
import { textBox } from './TextInputStyles'


// This is a reusable text input
const AppTextInput = ({ description, ...otherProps }) => {
    return (
        <View style={textBox.container}>
            <Text style={textBox.text}>{description}</Text>
            <TextInput style={textBox.textInput} {...otherProps} />
        </View>
    )
}

export default AppTextInput
