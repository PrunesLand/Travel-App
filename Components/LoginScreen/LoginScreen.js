import React, {useState} from 'react'
import { View, Text, ImageBackground, TouchableHighlight , TextInput} from 'react-native'
import {styles} from '../WelcomeScreen/WelcomeStyles'
import {logoStyle,RegStyles,square} from '../RegisterScreen/RegisterStyles'
import {MaterialCommunityIcons} from '@expo/vector-icons'
import RegLogButton from '../RegisterScreen/RegLogButton'
import {Login} from './LoginStyles'
import AppTextInput from '../TextInput/AppTextInput'
import * as yup from 'yup';
import {Formik} from 'formik'
// import {useNavigation} from '@react-navigation/native'


// export const LoginExit = ({onPress}) => {
//     return(
        
//         <View style={RegStyles.closeContainer}>
//         <TouchableHighlight onPress={onPress}>
//             <MaterialCommunityIcons
//             name='close'
//             style={RegStyles.close}
//             />
//         </TouchableHighlight>
//         </View>
//     )
// }

let schema = yup.object().shape({
    
    username: yup.string().required(),
    password: yup.string().required()
   
  });

const users=[
      {
          username:'John Jones',
          email:'jojo@g.com',
          password:'pass'
      },
      {
        username:'Billy Bat',
        email:'Billy@g.com',
        password:'pass4'
    }

];

const validateUser = ({username, password}) => {
    return(
        users.filter((user) => user.username === username && user.password === password).length>0
    );
};

const LoginScreen = ({navigation}) => {

    

    return (
        <View style={styles.container}>
            <ImageBackground 
            source={require("../Images/backgroundImg.jpg")}
            style={styles.background}
            />
            {/* <LoginExit 
            onPress={() => navigation.goBack()}
            /> */}
            <View style={styles.content}>
                {/* <View style={Login.logoContainer}>
                    <View style={styles.Logo}>
                        <MaterialCommunityIcons 
                            name='airplane'
                            size={130}
                            color='#fff'
                            />
                    </View>
                </View> */}
                
                <View style={Login.LoginContainer}>
                        <View style={Login.square}>
                            <Text 
                            style={RegStyles.title}
                            >Login</Text>

                            <Formik
                                initialValues={{email:'', password:'',username:''}}
                                onSubmit = {(values, {resetForm})=> {
                                    if(validateUser(values)){   
                                        resetForm();
                                        navigation.navigate('account', {
                                            screen: 'account',
                                            param:{
                                                screen:'account',
                                                param:{
                                                    paramName: values.username
                                                },
                                            }
                                        }
                                        );
                                        // createUser(values);
                                    }
                                    else{
                                        resetForm();
                                        alert("Invalid Login Details")
                                    }
                                }}
                                validationSchema={schema}
                            >

                                {({handleChange, handleSubmit, errors, setFieldTouched, touched, values}) =>(
                                    <>
                                    <AppTextInput 
                                    description='Name:' 
                                    clearTextOnFocus={true}
                                    onChangeText = {handleChange('username')}
                                    onBlur={()=> setFieldTouched('username')}
                                    value = {values.username}
                                    />
                                    {touched.username && <Text style={{color:'#fff', fontSize:15}}>{errors.username}</Text>}
                                    <AppTextInput 
                                    description='Password:' 
                                    secureTextEntry={true}
                                    clearTextOnFocus={true}
                                    onChangeText = {handleChange('password')}
                                    onBlur={()=> setFieldTouched('password')}
                                    value = {values.password}
                                    />
                                     { touched.password && <Text style={{color:'#fff', fontSize:15}}>{errors.password}</Text>}
                                    <RegLogButton title='Login' 
                                    onPress={ handleSubmit}
                                    />
                                    </>
                                    
                                )}

                            </Formik>

                            {/* <AppTextInput 
                            description='Name:' 
                            clearTextOnFocus={true}
                            />
                            <AppTextInput 
                            description='Password:' 
                            secureTextEntry={true}
                            clearTextOnFocus={true}
                            />
                            <RegLogButton title='Login' 
                            onPress={() => navigation.navigate('account')}
                            /> */}
                        </View>
                </View>
            </View>
        </View>
    )
}

export default LoginScreen
