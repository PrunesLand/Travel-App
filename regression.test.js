import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from './Components/LoginScreen/LoginScreen';
import RegLogButton from './Components/RegisterScreen/RegLogButton';
import AppTextInput from './Components/TextInput/AppTextInput';




it('Login screen renders', () => {
    const json = renderer
        .create(<LoginScreen />)
        .toJSON();
    expect(json).toMatchSnapshot();
});

it('Apptext input is rendered', () => {
    const item = renderer.create(<AppTextInput />).toJSON;
    expect(item).toMatchSnapshot();
})

