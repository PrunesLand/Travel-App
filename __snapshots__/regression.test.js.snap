// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apptext input is rendered 1`] = `[Function]`;

exports[`Login screen renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Components/Images/backgroundImg.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(45, 106, 79, 0.9)",
            "borderRadius": 15,
            "height": 450,
            "left": 10,
            "top": 100,
            "width": 370,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 40,
            }
          }
        >
          Login
        </Text>
        <View
          style={
            Object {
              "alignContent": "center",
              "height": 80,
              "paddingBottom": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "padding": 5,
              }
            }
          >
            Name:
          </Text>
          <TextInput
            allowFontScaling={true}
            clearTextOnFocus={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "color": "#000",
                "fontSize": 15,
                "height": 45,
                "padding": 10,
                "width": 300,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "height": 80,
              "paddingBottom": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "padding": 5,
              }
            }
          >
            Password:
          </Text>
          <TextInput
            allowFontScaling={true}
            clearTextOnFocus={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "color": "#000",
                "fontSize": 15,
                "height": 45,
                "padding": 10,
                "width": 300,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B4332",
                "borderRadius": 15,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "top": 20,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                }
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Components/Images/backgroundImg.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(45, 106, 79, 0.9)",
            "borderRadius": 15,
            "height": 450,
            "left": 10,
            "top": 100,
            "width": 370,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 40,
            }
          }
        >
          Login
        </Text>
        <View
          style={
            Object {
              "alignContent": "center",
              "height": 80,
              "paddingBottom": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "padding": 5,
              }
            }
          >
            Name:
          </Text>
          <TextInput
            allowFontScaling={true}
            clearTextOnFocus={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "color": "#000",
                "fontSize": 15,
                "height": 45,
                "padding": 10,
                "width": 300,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "height": 80,
              "paddingBottom": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "padding": 5,
              }
            }
          >
            Password:
          </Text>
          <TextInput
            allowFontScaling={true}
            clearTextOnFocus={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "color": "#000",
                "fontSize": 15,
                "height": 45,
                "padding": 10,
                "width": 300,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B4332",
                "borderRadius": 15,
                "display": "flex",
                "height": 50,
                "justifyContent": "center",
                "top": 20,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                }
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
